2-collectors/scala-stream-collector/kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -128,6 +142,29 @@ object KinesisSink {
     } catch {
       case _: ResourceNotFoundException => false
     }
+
+  private def createSqsClient(provider: AWSCredentialsProvider, region: String) =
+    Try(
+      AmazonSQSClientBuilder
+        .standard()
+        .withRegion(region)
+        .withCredentials(provider)
+        .build
+    ).toEither
+
+  def sqsBuffer(
+    sqsBufferName: Option[String],
+    provider: AWSCredentialsProvider,
+    region: String
+  ): Either[Throwable, Option[SqsClientAndName]] =
+    sqsBufferName match {
+      case Some(name) =>
+        for {
+          amazonSqs <- createSqsClient(provider, region)
```

Comment:
I'd use flatMap to make it more concise

Reply:
Thanks, I had two lines in for comp in the beginning, so it's kind of a leftover. Now even `map` is enough inside! Pattern match is needed on the outside, as None should be successful.